akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1230,15 +1391,28 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
 
 /**
  * An asynchronous callback holder that is attached to a [[GraphStageLogic]].
- * Invoking [[AsyncCallback#invoke]] will eventually lead to the registered handler
+ * Initializing [[AsyncCallback#invoke]] will eventually lead to the registered handler
  * being called.
+ *
+ * This holder has the same lifecycle as a stream and cannot be used before
+ * materialization is done.
+ *
+ * Typical use cases are exchanging messages between stream and substreams or invoking from external world sending
+ * event to a stream
  */
 trait AsyncCallback[T] {
   /**
-   * Dispatch an asynchronous notification. This method is thread-safe and
+   * Dispatch an asynchronous notification. This method must be thread-safe and
    * may be invoked from external execution contexts.
    */
   def invoke(t: T): Unit
+  /**
+   * Dispatch an asynchronous notification. This method must ve thread-safe and
```

Comment:
Newline after the first sentence, since first line shows up in summary of scaladoc AFAIR?

Reply:
"This method must ve thread-safe and" => "This method is thread-safe and"